speechcolab/datasets/gigaspeech.py

```diff
@@ -1,15 +1,48 @@
 from pathlib import Path
-
-allowed_subsets = ('XS', 'S', 'M', 'L', 'XL')
+import ijson
 
 
 class GigaSpeech(object):
     def __init__(self, path='.'):
-        self.path = path
+        self.root_path = Path(path)
+        self.json_path = self.root_path / 'GigaSpeech.json'
+        self.allowed_subsets = ('{XS}', '{S}', '{M}', '{L}', '{XL}')
+
+    def download(self, subset='{XL}'):
+        self.root_path.mkdir(parents=True, exist_ok=True)
+
+        assert subset in self.allowed_subsets, \
+            f'subset {subset} not in {self.allowed_subsets}'
+
+        # TODO: download the data
+        pass
```

Comment:
I think it might be a good idea if you print out some warning message, or raise NotImplementedError exception.